src/hera/validators.py

```diff
@@ -21,8 +21,10 @@ def validate_name(name: str, max_length: Optional[int] = None) -> str:
     if "_" in name:
         raise ValueError("Name cannot include an underscore")
 
-    pattern = r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
-    match_obj = re.match(pattern, name)
+    pattern = r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+    if generate_name:
+        pattern += r"(\.?|-*)"
```

Comment:
Does this need a quantifier at the end?
```suggestion
        pattern += r"(\.?|-*)*"
```

Reply:
No, that would allow the name to end in multiple consecutive dots, which is invalid. There should always be at least one character between two dots (DNS naming restriction).